src/Microsoft.ML.Data/Utils/ApiUtils.cs

```diff
@@ -23,7 +23,8 @@ private static OpCode GetAssignmentOpCode(Type t)
             if (t == typeof(ReadOnlyMemory<char>) || t == typeof(string) || t.IsArray ||
                 (t.IsGenericType && t.GetGenericTypeDefinition() == typeof(VBuffer<>)) ||
                 (t.IsGenericType && t.GetGenericTypeDefinition() == typeof(Nullable<>)) ||
-                t == typeof(DateTime) || t == typeof(DateTimeOffset) || t == typeof(TimeSpan) || t == typeof(DataViewRowId))
+                t == typeof(DateTime) || t == typeof(DateTimeOffset) || t == typeof(TimeSpan) ||
```

Comment:
>DateTimeOffset [](start = 53, length = 14)

Should date*  types go into TypeManager?  They are structs, I guess.

Reply:
I don't want to mix users types with ML.NET's native types.

---
In reply to: [284939178](https://github.com/dotnet/machinelearning/pull/3263#discussion_r284939178) [](ancestors = 284939178)